onmt/inputters/dataset_base.py

```diff
@@ -87,13 +80,6 @@ def __init__(self, fields, src_examples_iter, tgt_examples_iter,
 
         super(DatasetBase, self).__init__(examples, fields, filter_pred)
 
-    @staticmethod
-    def _raise_missing_dep(*missing_deps):
-        """Raise missing dep exception with standard error message."""
-        raise MissingDependencyException(
-            "Could not create reader. Be sure to install "
-            "the following dependencies: " + ", ".join(missing_deps))
-
     def __getattr__(self, attr):
```

Comment:
This method predates this pull request, but what is it for? `self.fields` should be defined from the moment the constructor is done.

Reply:
(Specifically it's from #1188) Those two lines are necessary. It's for the unpickling side. Pickle does a ``getattr`` call to try to find ``__setstate__``. Since the state is not yet set, ``self.fields`` actually isn't defined. I got it from [here](https://stackoverflow.com/a/49380669/8217456).

Reply:
Ah, understood.